### PR TITLE
refactor: change isAuthenticated and validateWalletId to be deterministic

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR organizes the graphql queries and mutation by authentication status and walletId validation before passing into the schema creation method. These same queries/mutations are then used to populate shields and middleware, instead of having to manually maintain these.

### CODE REVIEW

File: src/graphql/admin/mutations.ts [hunk 1]
```diff
@@ -12,23 +12,25 @@ import ColdStorageRebalanceToHotWalletMutation from "@graphql/admin/root/mutatio
...
     coldStorageRebalanceToHotWallet: ColdStorageRebalanceToHotWalletMutation,
-  }),
-})
+  },
+}
 
-export default MutationType
+export const MutationType = GT.Object({
+  name: "Mutation",
+  fields: () => ({ ...mutationFields.unauthed, ...mutationFields.authed }),
```
Comment (nicolasburtey) [inline src/graphql/admin/mutations.ts:35]: this is a great idea. force us to make sure we don't have an authenticated endpoint without putting it here!

File: src/graphql/admin/mutations.ts [hunk 2]
```diff
@@ -12,23 +12,25 @@ import ColdStorageRebalanceToHotWalletMutation from "@graphql/admin/root/mutatio
...
-  fields: () => ({
+export const mutationFields = {
+  unauthed: {
     userRequestAuthCode: UserRequestAuthCodeMutation,
     userLogin: UserLoginMutation,
 
     captchaCreateChallenge: CaptchaCreateChallengeMutation,
     captchaRequestAuthCode: CaptchaRequestAuthCodeMutation,
 
+    accountsAddUsdWallet: AccountsAddUsdWalletMutation,
```
Comment (nicolasburtey) [inline src/graphql/admin/mutations.ts:23]: should probably be authed

File: src/graphql/main/mutations.ts [hunk 3]
```diff
@@ -33,55 +33,63 @@ import CaptchaRequestAuthCodeMutation from "@graphql/root/mutation/captcha-reque
...
+    lnUsdInvoiceCreateOnBehalfOfRecipient: LnUsdInvoiceCreateOnBehalfOfRecipientMutation,
+    lnNoAmountInvoiceCreateOnBehalfOfRecipient:
+      LnNoAmountInvoiceCreateOnBehalfOfRecipientMutation,
+  },
 
-  // authed
-  userQuizQuestionUpdateCompleted: UserQuizQuestionUpdateCompletedMutation,
-  deviceNotificationTokenCreate: DeviceNotificationTokenCreateMutation,
+  authed: {
+    noWalletId: {
```
Comment (nicolasburtey) [inline src/graphql/main/mutations.ts:51]: could also be `account` and `wallet` as a way to seggregate those two set of mutations

**Review by nicolasburtey — DISMISSED** [review]

**Comment by vindard** [comment]
Note: this [`force-push`](https://github.com/GaloyMoney/galoy/compare/50730f6f873e3d535589e7547a8233662a6d6ca6..91c9743651b9e89932ef7a0bbe450277f17fc0b8) was to fix that admin schema was being wrongly changed when we ran `make start-main` locally

File: src/graphql/main/mutations.ts (hunk 3 above)
Comment (vindard) [reply src/graphql/main/mutations.ts]: I was thinking this as well when I was moving these around. Done https://github.com/GaloyMoney/galoy/pull/2097/commits/5bc7ab6fd86590e6f410a73f4cc3988d26d61c5e

Comment (vindard) [reply src/graphql/main/mutations.ts]: Done https://github.com/GaloyMoney/galoy/pull/2097/commits/b973aea69d3903206f09a5942c6a5c533ed8130d

File: src/graphql/admin/mutations.ts (hunk 2 above)
Comment (vindard) [reply src/graphql/admin/mutations.ts]: I was thinking this as well when I was moving these around. Done https://github.com/GaloyMoney/galoy/pull/2097/commits/5bc7ab6fd86590e6f410a73f4cc3988d26d61c5e

**Review by nicolasburtey — APPROVED** [review]

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**